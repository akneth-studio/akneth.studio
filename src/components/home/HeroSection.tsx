import CTAButton from "../CTAButton";

export default function HeroSection() {
  return (
    <section className="hero text-center py-4">
      <div id='lead-gradient' className='px-4 mb-4'>
        <h1>Szablony, automatyzacje i indywidualne rozwiązania<br/>
        dla firm i klientów indywidualnych.</h1><br />
        <p className='lead fs-1'>Elastyczność, uczciwość i szybka realizacja - Twoje potrzeby, mój priorytet.</p>
      </div>
      <CTAButton
        type='button'
        text="Skontaktuj się"
        variant="primary"
        to="/contact"
        size='lg'
      />
    </section>
  );
}
