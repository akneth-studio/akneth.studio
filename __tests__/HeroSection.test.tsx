import React from 'react';
import { render, screen } from '@testing-library/react';
import HeroSection from '../src/components/home/HeroSection';
import '@testing-library/jest-dom';

describe('HeroSection component', () => {
  it('renders heading, paragraph and CTA button', () => {
    render(<HeroSection />);
    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();
    expect(screen.getByText(/szablony, automatyzacje i indywidualne rozwiązania/i)).toBeInTheDocument();
    expect(screen.getByText(/elastyczność, uczciwość i szybka realizacja/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /skontaktuj się/i })).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<HeroSection />);
    expect(container).toMatchSnapshot();
  });
});
