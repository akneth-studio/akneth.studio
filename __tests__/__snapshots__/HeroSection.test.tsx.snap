// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HeroSection component matches snapshot 1`] = `
<div>
  <section
    class="hero text-center py-4"
  >
    <div
      class="px-4 mb-4"
      id="lead-gradient"
    >
      <h1
        class="shinydark"
      >
        Szablony, automatyzacje i indywidualne rozwiązania
        <br />
        dla firm i klientów indywidualnych.
      </h1>
      <br />
      <p
        class="lead fs-1"
      >
        Elastyczność, uczciwość i szybka realizacja - Twoje potrzeby, mój priorytet.
      </p>
    </div>
    <button
      aria-label="Skontaktuj się"
      class="btn btn-primary btn-lg  btn btn-primary"
      data-testid="cta-button"
      role="button"
      tabindex="0"
      type="button"
    >
      Skontaktuj się
    </button>
  </section>
</div>
`;
